### PR TITLE
[IMP] MRP : new settings - Making changes for new mrp settings

### PR DESCRIPTION
**Pull request by pgu-odoo**
Description of the issue/feature this PR addresses:
https://www.odoo.com/web#id=29529&view_type=form&model=project.task&action=327&menu_id=

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/mrp/views/mrp_config_settings_views.xml [hunk 1]
```diff
@@ -10,21 +10,140 @@
...
+                                    Go through a production routing with work orders
+                                </div>
+                            </div>
+                        </div>
+                        <div class="col-md-6 o_setting_box" id="working_time" groups="mrp.group_mrp_routings">
+                            <div class="o_setting_left_pane">
+                                <field name="working_time"/>
+                            </div>
+                            <div class="o_setting_right_pane">
+                                <span class="fa fa-2x" title="Working Time"/>
```
Comment (mdi-odoo) [inline addons/mrp/views/mrp_config_settings_views.xml]: Why ?

File: addons/web/static/src/js/views/form_upgrade_widgets.js [hunk 2]
```diff
@@ -23,7 +23,7 @@ var AbstractFieldUpgrade = {
         this._super.apply(this, arguments);
         this.get_enterprise_label().after($("<span>", {
             text: "Enterprise",
-            'class': "label label-primary oe_inline"
+            'class': "label label-primary oe_inline oe_enterprise_label"
```
Comment (mdi-odoo) [inline addons/web/static/src/js/views/form_upgrade_widgets.js]: Remove this change.

File: addons/mrp/views/mrp_config_settings_views.xml [hunk 3]
```diff
@@ -10,21 +10,140 @@
...
+                            <div class="o_setting_left_pane">
+                                    <field name="group_product_variant"/>
+                            </div>
+                            <div class="o_setting_right_pane">
+                                <label for="group_product_variant"/>
+                                <div class="text-muted">
+                                    Set product attributes (e.g. color, size) to manage variants
+                                </div>
+                                <div class="mt16" attrs="{'invisible': [('group_product_variant','=',False)]}">
+                                    <button name="%(product.attribute_action)d" icon="fa-arrow-right" type="action" string="Attributes" class="btn-link o_btn_link_padding0"/>
```
Comment (mdi-odoo) [inline addons/mrp/views/mrp_config_settings_views.xml]: Same as above.

File: addons/web/static/src/less/form_view.less [hunk 4]
```diff
@@ -502,3 +502,41 @@
     float: left; // Unfortunately needed for the bounce effect
     margin-right: 4px;
 }
+// Light labels (not in bold) for settings forms
+label.oe_light_label, .oe_light_label label {
```
Comment (mdi-odoo) [inline addons/web/static/src/less/form_view.less]: Move this change to **form_view.less** .

File: addons/mrp/views/mrp_config_settings_views.xml [hunk 5]
```diff
@@ -10,21 +10,139 @@
...
+                                <div class="text-muted">
+                                    Go through a production routing with work orders
+                                </div>
+                            </div>
+                        </div>
+                        <div class="col-md-6 o_setting_box" id="working_time" groups="mrp.group_mrp_routings">
+                            <div class="o_setting_left_pane">
+                                <field name="working_time"/>
+                            </div>
+                            <div class="o_setting_right_pane" title="Working Time">
```
Comment (mdi-odoo) [inline addons/mrp/views/mrp_config_settings_views.xml]: Remove title="Working Time" .

File: addons/mrp/views/mrp_config_settings_views.xml [hunk 6]
```diff
@@ -10,21 +10,139 @@
...
+                                <field name="manufacturing_lead_boolean"/>
+                            </div>
+                            <div class="o_setting_right_pane">
+                                <label for="manufacturing_lead_boolean"/>
+                                <span class="fa fa-2x fa-building-o pull-right" title="Values set here are company-specific." groups="base.group_multi_company"/>
+                                <div class="text-muted">
+                                        Schedule manufacturing orders earlier to avoid delays
+                                 </div>
+                                <div class="mt16" attrs="{'invisible': [('manufacturing_lead_boolean','=',False)]}">
+                                    <div class="text-muted oe_inline">
```
Comment (mdi-odoo) [inline addons/mrp/views/mrp_config_settings_views.xml]: Remove class="oe_inline" .

File: addons/web/static/src/less/form_view.less [hunk 7]
```diff
@@ -502,3 +502,41 @@
...
+            padding-left: 10px;
+            .o_form_input_dropdown > .o_form_input {
+                width: 100%;
+            }
+            .o_form_input {
+                width: 50%;
+            }
+            button.btn-link {
+                padding: 0;
+            }
```
Comment (mdi-odoo) [inline addons/web/static/src/less/form_view.less]: Please move CSS related changes to **form_view.less** .

File: addons/mrp/views/mrp_config_settings_views.xml [hunk 8]
```diff
@@ -10,21 +10,149 @@
...
+                            </div>
+                        </div>
+                        <div class="col-md-6 o_setting_box" id="mrp_byproduct">
+                            <div class="o_setting_left_pane">
+                                <field name="module_mrp_byproduct"/>
+                            </div>
+                            <div class="o_setting_right_pane">
+                                <label for="module_mrp_byproduct"/>
+                                <div class="text-muted">
+                                    Add by-products to bills of materials. This can be used to get several finished products as well. Without this option you only do: A + B = C. With the option: A + B = C + D.
```
Comment (mdi-odoo) [inline addons/mrp/views/mrp_config_settings_views.xml]: Remove this and just add this statement as a tooltip .

File: addons/mrp/views/mrp_config_settings_views.xml [hunk 9]
```diff
@@ -7,24 +7,156 @@
...
+                                </div>
+                            </div>
+                        </div>
+                        <div class="col-md-6 col-xs-12 o_setting_box" id="security_lead_time">
+                            <div class="o_setting_left_pane">
+                                <field name="manufacturing_lead_boolean"/>
+                            </div>
+                            <div class="o_setting_right_pane">
+                                <label for="manufacturing_lead_boolean"/>
+                                <span class="fa fa-2x fa-building-o pull-right" title="Values set here are company-specific." groups="base.group_multi_company"/>
```
Comment (mdi-odoo) [inline addons/mrp/views/mrp_config_settings_views.xml]: Move the multi-company icon to left side and also change the size to fa-lg .

File: addons/mrp/models/mrp_config_settings.py [hunk 10]
```diff
@@ -1,7 +1,7 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import fields, models
+from odoo import api, fields, models
```
Comment (mdi-odoo) [inline addons/mrp/models/mrp_config_settings.py]: Please remove unused import statetment.

File: addons/mrp/models/mrp_config_settings.py [hunk 11]
```diff
@@ -10,39 +10,16 @@ class MrpConfigSettings(models.TransientModel):
 
     company_id = fields.Many2one('res.company', string='Company', required=True,
         default=lambda self: self.env.user.company_id)
-    manufacturing_lead = fields.Float(related='company_id.manufacturing_lead', string="Manufacturing Lead Time *")
-    group_product_variant = fields.Selection([
-        (0, "No variants on products"),
-        (1, 'Products can have several attributes, defining variants (Example: size, color,...)')], "Product Variants",
-        help='Work with product variant allows you to define some variant of the same products, an ease the product management in the ecommerce for example',
+    manufacturing_lead = fields.Float(related='company_id.manufacturing_lead', string="Manufacturing Lead Time")
+    default_manufacturing_lead = fields.Boolean(string="Default Manufacturing Lead Time", default_model='mrp.config.settings')
```
Comment (mdi-odoo) [inline addons/mrp/models/mrp_config_settings.py]: Rename **default_manufacturing_lead** to **default_use_manufacturing_lead**.

File: addons/mrp/models/mrp_config_settings.py [hunk 12]
```diff
@@ -10,39 +10,16 @@ class MrpConfigSettings(models.TransientModel):
...
+    default_working_time = fields.Boolean(string="Working Time", default_model='mrp.config.settings')
```
Comment (mdi-odoo) [inline addons/mrp/models/mrp_config_settings.py]: Rename **default_working_time** to **default_use_working_time** .

File: addons/maintenance/models/maintenance.py [hunk 13]
```diff
@@ -39,7 +39,7 @@ def _compute_fold(self):
     maintenance_ids = fields.One2many('maintenance.request', 'category_id', copy=False)
     maintenance_count = fields.Integer(string="Maintenance", compute='_compute_maintenance_count')
     alias_id = fields.Many2one(
-        'mail.alias', 'Alias', ondelete='cascade', required=True,
+        'mail.alias', 'Alias', ondelete='restrict', required=True,
```
Comment (tivisse) [inline addons/maintenance/models/maintenance.py]: Why ?

Comment (tivisse) [inline addons/maintenance/models/maintenance.py]: Ok I got it :)

Comment (atp-odoo) [inline addons/maintenance/models/maintenance.py]: when we uninstall mrp_maintenance module , due to ondelete='cascade' and due to this reference in mrp_maintencance  demo record : field name="alias_id" ref="maintenance.mail_alias_equipment" 
it's remove record from maintenance module. 
then when we try to install again mrp_maintenance we got traceback : external id reference not found.. (maintenance.mail_alias_equipment)

**Pull request merged**